promoter/promoter.go

```diff
@@ -186,13 +193,90 @@ func (p *Promoter) staticAddrDiff(ctx context.Context) (toAdd []WatchedAddress,
 // threadedCreditTransactions continuously polls the db for uncreditted txns and
 // notifies the credit promoter about them.
 func (p *Promoter) threadedCreditTransactions() {
-	// TODO: Fetch all txns with credited == false.
+	t := time.NewTicker(txnPollInterval)
+	defer t.Stop()
+LOOP:
+	for {
+		select {
+		case <-p.staticBGCtx.Done():
+			return
+		case <-t.C:
+		}
+
+		// Loop over txns one-by-one.
+		for {
+			// Fetch an transaction that the credit system doesn't know
+			// about yet.
+			currentTime := time.Now().UTC()
+			sr := p.staticColTransactions().FindOneAndUpdate(p.staticBGCtx, bson.M{
+				"credited_at": bson.M{
+					"$lt": currentTime.Add(-txnPollInterval),
+				},
+			}, bson.M{
+				"$set": bson.M{
+					"credited_at": currentTime,
+				},
+			}, options.FindOneAndUpdate())
```

Comment:
Just FYI: the options param is optional, you can skip it when you don't need to pass custom options.